src/map.rs

```diff
@@ -2367,6 +2371,63 @@ where
     }
 }
 
+impl<'de, K, V> Deserialize<'de> for HashMap<K, V, crate::DefaultHashBuilder>
+where
+    K: 'static + Deserialize<'de> + Send + Sync + Hash + Clone + Eq,
+    V: 'static + Deserialize<'de> + Send + Sync + Eq,
+{
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_map(HashMapVisitor::new())
+    }
+}
+
+struct HashMapVisitor<K, V> {
+    key_marker: PhantomData<K>,
+    value_marker: PhantomData<V>,
+}
+
+impl<K, V> HashMapVisitor<K, V> {
+    pub(crate) fn new() -> Self {
+        Self {
+            key_marker: PhantomData,
+            value_marker: PhantomData,
+        }
+    }
+}
+
+impl<'de, K, V> Visitor<'de> for HashMapVisitor<K, V>
+where
+    K: 'static + Deserialize<'de> + Send + Sync + Hash + Clone + Eq,
+    V: 'static + Deserialize<'de> + Send + Sync + Eq,
+{
+    type Value = HashMap<K, V, crate::DefaultHashBuilder>;
+
+    fn expecting(&self, _f: &mut Formatter<'_>) -> fmt::Result {
+        todo!("Create an error message");
+    }
+
+    fn visit_map<M>(self, mut access: M) -> Result<Self::Value, M::Error>
+    where
+        M: MapAccess<'de>,
+    {
+        let map = HashMap::with_capacity(access.size_hint().unwrap_or(0));
```

Comment:
This should probably call `new` if `size_hint().is_none()`, rather than call `with_capacity(0)`. The two are not guaranteed to be the same.

Reply:
Alright. Fixed in the last commit.